src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let status = Object.assign({}, state);
+  const result = [];
+
+  actions.forEach(action => {
+    let newState = {};
+
+    if (action.type === 'addProperties') {
+      newState = Object.assign({}, status, action.extraData);
+      Object.assign({}, status, action.extraData);
```

Comment:
With those simple change above, you can just do like this:
```suggestion
      Object.assign(stateCopy, action.extraData);
```

Reply:
Btw, you could destructure properties of the `action` at the top of the loop